doc/type_adaption.md

```diff
@@ -116,6 +116,43 @@ we need to go from any other node in the tree.
 #### Finding no hierarchy
 If we can not find any hierarchy we currently return the input type unchanged.
 
+#### Adapting generics of enclosing classes
+Java allows classes to use generics of their enclosing class, e.g.
+```java
+public class Outer<A> {
+  public static class Inner<B> {
+    public void bar(A a, B b) {}
+  }
+}
```

Comment:
This doesn't compile for me, I assume the inner class shouldn't be static?

Reply:
Whoops, you are right.